edit-post/components/header/plugins/index.js

```diff
@@ -0,0 +1,100 @@
+/**
+ * External dependencies
+ */
+import { map, isEmpty, isString } from 'lodash';
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { withInstanceId, IconButton, MenuItemsGroup } from '@wordpress/components';
+import { withSelect } from '@wordpress/data';
+import { compose } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import './style.scss';
+import { getMoreMenuItems } from '../../../api/more-menu-item';
+
+/**
+ * Renders a list of plugins that will activate different UI elements.
+ *
+ * @param {Object} props The component props.
+ *
+ * @return {Object} The rendered list of menu items.
+ */
+function Plugins( props ) {
+	const ellipsisMenuItems = getMoreMenuItems();
+
+	if ( isEmpty( ellipsisMenuItems ) ) {
+		return null;
+	}
+
+	/**
+	 * Handles the user clicking on one of the plugins in the menu
+	 *
+	 * @param {string} pluginId The plugin id.
+	 *
+	 * @return {void}
+	 */
+	function onSelect( pluginId ) {
+		props.onSelect();
+		ellipsisMenuItems[ pluginId ].callback();
+	}
+
+	return (
+		<MenuItemsGroup
+			label={ __( 'Plugins' ) }
+			filterName="editPost.MoreMenu.plugins" >
+			{ map( ellipsisMenuItems, menuItem => {
+				if ( isString( menuItem.icon ) ) {
+					menuItem.icon = null;
```

Comment:
You're mutating the registered value of `menuItem` here. Are you intending to? Shouldn't that occur at registration-time, not in the render? Renders shouldn't have side effects.